devicemodel/hw/pci/wdt_i6300esb.c

```diff
@@ -41,15 +41,15 @@
 #define ESB_WDT_REBOOT	(0x01 << 5)   /* Enable reboot on timeout */
 #define ESB_WDT_RELOAD	(0x01 << 8)   /* Ping/kick dog  */
 #define	ESB_WDT_TIMEOUT	(0x01 << 9)   /* WDT timeout happened? */
-#define TIMER_TO_SECONDS(val)	(val >> 9)
+#define TIMER_TO_SECONDS(val)	(val >> 10) /* 10bits for 1s in a 1KHz Clock */
```

Comment:
This is still confusing. How 1KHz is reflected here?

Reply:
10bits for 1s means the unit is (1s / 2^^10)

Reply:
this is defined by i6300 HW spec that the default prescaler is 1KHz clock for a 20-bit timer counter. Because 2^^10 means 1s, so the counter value right shift 10 bits could get how many seconds.

how about below:

/* The default prescaler is 1KHz clock for a 20-bit counter per i6300esb spec,
 * so right shift 10 bits of timer counter to get the exact seconds for this timer.
 */
#define TIMER_TO_SECONDS(val) (val >> 10)

Reply:
prescaler is just a scaler, it is not a frequency.  Original Frequency * pre_scaler becomes the real counter frequency.

Reply:
/* per i6300esb spec:
 * The 20-bit Preload Value is loaded into bits 34:15 of the
 * main down counter. The resulting timer clock is the PCI
 * Clock (33 MHz) divided by 2^^15 . The approximate clock
 * generated is 1 KHz, so right shift 10 bits of preload value
 * to get the exact seconds for this timer.
 */
#define TIMER_TO_SECONDS(val) (val >> 10)